sources/packages/api/src/route-controllers/assessment/models/assessment.system-access.dto.ts

```diff
@@ -0,0 +1,168 @@
+  IsNotEmptyObject,
+  IsNumber,
+  IsOptional,
+  IsPositive,
+  IsUUID,
+  Min,
+  ValidateNested,
+} from "class-validator";
+import {
+  ProgramInfoStatus,
+  ApplicationData,
+  AssessmentTriggerType,
+  DisbursementValueType,
+  AssessmentStatus,
+} from "../../../database/entities";
+
+export class ProgramYearDetails {
+  programYear: string;
+  startDate: Date;
+  endDate: Date;
+}
+
+export class ApplicationOfferingDetails {
+  id: number;
+  studyStartDate: Date;
+  studyEndDate: Date;
+  actualTuitionCosts: number;
+  programRelatedCosts: number;
+  mandatoryFees: number;
+  exceptionalExpenses: number;
+  tuitionRemittanceRequestedAmount: number;
+  offeringDelivered: string;
+  offeringIntensity: string;
+}
+
+export class ApplicationProgramDetails {
+  programCredentialType: string;
+  programLength: string;
+}
+
+export class ApplicationInstitutionDetails {
+  institutionType: string;
+}
+
+export class ApplicationLocationDetails {
+  institutionLocationProvince: string;
+}
+
+export class ApplicationStudentDetails {
+  studentPDStatus?: boolean;
+}
+
+/**
+ * Assessment and application information used as a main
+ * source of data for assessment workflow process.
+ */
+export class ApplicationAssessmentDTO {
+  /**
+   * Origin of the assessment.
+   */
+  triggerType: AssessmentTriggerType;
+  /**
+   * Application dynamic data.
+   */
+  data: ApplicationData;
+  /**
+   * Details of the program year associated with the student application.
+   */
+  programYear: ProgramYearDetails;
+  /**
+   * Offering details for student application.
+   */
+  offering: ApplicationOfferingDetails;
+  /**
+   * Program details for student application.
+   */
+  program: ApplicationProgramDetails;
+  /**
+   * Institution details for student application.
+   */
+  institution: ApplicationInstitutionDetails;
+  /**
+   * Location details for student application.
+   */
+  location: ApplicationLocationDetails;
+  /**
+   * Student details for student application.
+   */
+  student: ApplicationStudentDetails;
+}
+
+export class UpdateProgramInfoDTO {
+  @IsOptional()
+  @IsPositive()
+  programId?: number;
+  @IsOptional()
+  @IsPositive()
+  offeringId?: number;
+  @IsPositive()
+  locationId: number;
+  @IsEnum(ProgramInfoStatus)
+  status: ProgramInfoStatus;
+}
+
+export class UpdateProgramInfoStatusDTO {
+  @IsEnum(ProgramInfoStatus)
+  status: ProgramInfoStatus;
+}
+
+export class UpdateAssessmentWorkflowIdDTO {
+  @IsUUID()
+  workflowId: string;
+}
+
+export class UpdateAssessmentDataDTO {
+  @IsNotEmptyObject()
+  data: any;
+}
+
+/**
+ * Values to be associated with a disbursement.
+ */
+export class DisbursementValueDTO {
+  @IsNotEmpty()
+  valueCode: string;
+  @IsEnum(DisbursementValueType)
+  valueType: DisbursementValueType;
+  @IsNumber()
+  @Min(0)
+  valueAmount: number;
+}
+
+/**
+ * Disbursement to be created altogether with its values.
+ */
+export class DisbursementScheduleDTO {
+  @IsNotEmpty()
+  disbursementDate: Date;
```

Comment:
for date fields we can use @IsDate() this allows datestring as well.

Reply:
This is a date only field 😉 
I tested and it worked. Change it!